packages/react/src/components/MultiSelect/index.tsx

```diff
@@ -0,0 +1,64 @@
+import React from 'react';
+import Select, { MultiValue, components, OptionProps } from 'react-select';
+import { Options } from '../../models/select';
+import * as Styles from './styles';
+import { Checkbox } from '../Checkbox';
+
+interface SelectProps {
+  options: Options[];
+  placeholder?: string;
+  name?: string;
+  isClearable?: boolean;
+  isSearchable?: boolean;
+  isDisabled?: boolean;
+  isLoading?: boolean;
+  closeMenuOnSelect?: boolean;
+  hideSelectedOptions?: boolean;
+}
+export interface MultiSelectProps {
+  setState: (value: string[]) => void;
+  select: SelectProps;
+  checkbox?: boolean;
+}
+
+const InputOption = ({ children, ...props }: OptionProps) => {
+  return (
+    <components.Option {...props}>
+      <Checkbox
+        root={{ checked: props.isSelected }}
+        text={{ label: children, classText: Styles.text() }}
+      />
+    </components.Option>
+  );
+};
+
+export const MultiSelect = ({
+  select,
+  setState,
+  checkbox,
+}: MultiSelectProps) => {
+  const onChange = (option: MultiValue<Options>) => {
```

Comment:
Não compensa só passar uma prop "onChange" e quem for fazer uso aplica essa lógica?